### PR TITLE
Refactor resource image and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_image.go [hunk 1]
```diff
@@ -360,17 +358,17 @@ func isIBMPIImageRefreshFunc(ctx context.Context, client *st.IBMPIImageClient, i
...
+		return image, Arg_ImageQueStatus, nil
 	}
 }
 
-func waitForIBMPIJobCompleted(ctx context.Context, client *st.IBMPIJobClient, jobID string, timeout time.Duration) (interface{}, error) {
+func waitForIBMPIJobCompleted(ctx context.Context, client *instance.IBMPIJobClient, jobID string, timeout time.Duration) (interface{}, error) {
 	stateConf := &resource.StateChangeConf{
-		Pending: []string{helpers.JobStatusQueued, helpers.JobStatusReadyForProcessing, helpers.JobStatusInProgress, helpers.JobStatusRunning, helpers.JobStatusWaiting},
-		Target:  []string{helpers.JobStatusCompleted, helpers.JobStatusFailed},
+		Pending: []string{Arg_JobStatusQueued, Arg_JobStatusReadyForProcessing, Arg_JobStatusInProgress, Arg_JobStatusRunning, Arg_JobStatusWaiting},
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_image.go:370]: So these aren't arguments to the terraform resource, and the value of them should not be changed. Make a new section for the status constants. Something like:

```
Status_Queued = "queued"
Status_InProgress = "in_progress"
...
```

File: ibm/service/power/resource_ibm_pi_image.go [hunk 2]
```diff
@@ -335,12 +333,12 @@ func resourceIBMPIImageDelete(ctx context.Context, d *schema.ResourceData, meta
...
 }
 
-func isWaitForIBMPIImageAvailable(ctx context.Context, client *st.IBMPIImageClient, id string, timeout time.Duration) (interface{}, error) {
+func isWaitForIBMPIImageAvailable(ctx context.Context, client *instance.IBMPIImageClient, id string, timeout time.Duration) (interface{}, error) {
 	log.Printf("Waiting for Power Image (%s) to be available.", id)
 
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"retry", helpers.PIImageQueStatus},
-		Target:     []string{helpers.PIImageActiveStatus},
+		Pending:    []string{"retry", Image_Questatus},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_image.go:340]: `PIImageQueStatus         = "queued"`

File: ibm/service/power/resource_ibm_pi_image.go [hunk 3]
```diff
@@ -335,12 +333,12 @@ func resourceIBMPIImageDelete(ctx context.Context, d *schema.ResourceData, meta
...
+		Target:     []string{Image_ActiveStatus},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_image.go:341]: `PIImageActiveStatus      = "active"`

File: ibm/service/power/resource_ibm_pi_image.go [hunk 4]
```diff
@@ -381,9 +379,9 @@ func waitForIBMPIJobCompleted(ctx context.Context, client *st.IBMPIJobClient, jo
 				log.Printf("[DEBUG] get job failed with empty response")
 				return nil, "", fmt.Errorf("failed to get job status for job id %s", jobID)
 			}
-			if *job.Status.State == helpers.JobStatusFailed {
+			if *job.Status.State == Status_Failed {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_image.go:382]: `JobStatusFailed             = "failed"`

**Review by michaelkad — CHANGES_REQUESTED** [review]
Const updated incorrectly.

**Comment by michaelkad** [comment]
Opened #130

**Pull request closed without merging**